pagespeed/system/system_test.sh

```diff
@@ -2249,3 +2249,10 @@ if [ "$CACHE_FLUSH_TEST" = "on" ]; then
   # check [ $bytes -gt 100000 ]
   check [ $bytes -lt 100000 ]
 fi
+
+start_test AddResourceHeaders works for pagespeed resources.
+URL="$TEST_ROOT/compressed/hello_js.custom_ext.pagespeed.ce.HdziXmtLIV.txt"
+HTML_HEADERS=$($WGET_DUMP $URL)
+# TODO(oschaaf): a helper to perform this type of matching.
+MATCHES=$(echo "$HTML_HEADERS" | grep -c "^X-Foo: Bar") || true
```

Comment:
I'll use check_from. Does check_from also ensure there is single instance of the header? I think that is the common case, and in nps we currently have, for example, multiple vary headers being emitted.